numpy/lib/recfunctions.py

```diff
@@ -523,6 +523,39 @@ def _drop_descr(ndtype, drop_names):
     return _fix_output(output, usemask=usemask, asrecarray=asrecarray)
 
 
+def _keep_fields(base, keep_names, usemask=True, asrecarray=False):
+    """
+    Return a new array keeping only the fields in `keep_names`,
+    and preserving the order of those fields.
+
+    Parameters
+    ----------
+    base : array
+        Input array
+    keep_names : string or sequence
+        String or sequence of strings corresponding to the names of the
+        fields to keep. Order of the names will be preserved.
+    usemask : {False, True}, optional
+        Whether to return a masked array or not.
+    asrecarray : string or sequence, optional
+        Whether to return a recarray or a mrecarray (`asrecarray=True`) or
+        a plain ndarray or masked array with flexible dtype. The default
+        is False.
+    """
+    if _is_string_like(keep_names):
+        uniq_names = [keep_names, ]
+    else:
+        uniq_names = []
+        for n in keep_names:
+            if n not in uniq_names:
+                uniq_names.append(n)
```

Comment:
Is there even any point in computing `uniq_names`? I get an error on my machine if I pass the same fieldname twice anyway